scripts/monitor/visualization/resource_monitoring.py

```diff
@@ -0,0 +1,596 @@
+from typing import Dict
+import matplotlib.pyplot as plt
+from bokeh.models import Title
+from matplotlib.backends.backend_pdf import PdfPages
+import numpy
+import pandas as pd
+import pandas_bokeh
+from operator import truediv
+
+
+class ResourceUsage:
+    instances = 0  # count number of class instances (can't find a way to update this when an instance is deleted)
+
+    def __init__(self, meta_runtime: pd.DataFrame, metrics: pd.DataFrame) -> None:
+        """
+        initialization step: bring meta, runtime, and metrics data into scope
+        :param meta_runtime: meta_runtime table
+        :param metrics: metrics table
+        """
+        self.meta_runtime = meta_runtime
+        self.metrics = metrics
+        self.instances += 1
```

Comment:
this is what I'm concerned about: it's almost always going to be 1, as this `instance` is not the class-level `instance`, and class level `instance` is never changed.

Reply:
oops this is a mistake, I used self out of habit since that's what I did for every other variable. Changed to ResourceUsage.instances now